packages/flutter/lib/src/material/tabs.dart

```diff
@@ -891,14 +899,14 @@ class TabBar extends StatefulWidget implements PreferredSizeWidget {
   /// [AppBar] uses this size to compute its own preferred size.
   @override
   Size get preferredSize {
+    double maxHeight = _kTabHeight;
     for (final Widget item in tabs) {
-      if (item is Tab) {
-        final Tab tab = item;
-        if ((tab.text != null || tab.child != null) && tab.icon != null)
-          return Size.fromHeight(_kTextAndIconTabHeight + indicatorWeight);
+      if (item is PreferredSizeWidget) {
+        final double itemHeight = item.preferredSize.height;
+        maxHeight = itemHeight > maxHeight ? itemHeight : maxHeight;
```

Comment:
This is a bit more readable:
```suggestion
        maxHeight = math.max(itemHeight,  maxHeight);
```

Reply:
The linux-analyze test seems to be failing while checking the package allowlist.